src/main/java/eu/europeana/metis/sandbox/service/debias/State.java

```diff
@@ -0,0 +1,15 @@
+package eu.europeana.metis.sandbox.service.debias;
+
+import eu.europeana.metis.sandbox.repository.debias.DetectRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class State {
+
+  public static final Logger LOGGER = LoggerFactory.getLogger(State.class);
+
+  protected DetectService stateMachine;
```

Comment:
Yes. I saw that. that some circular reference can happen, that is why only from the constructor the states are created for every (detect machine), and also removed the beans to prevent this leaving only the DebiasDetectService.